### PR TITLE
add contributing.md and associated documents

### PR DESCRIPTION
**Pull request by Remi-Gau**
This is meant to create a document to centralize all the "HOW TO" for people who want to contribute to this repo:

- for code style guide / quality
- for unit testing
- for dealing with pull requests...

In case you want to see it in an easier format: this is adapted from the one in the BIDS specs: https://github.com/bids-standard/bids-specification/blob/master/CONTRIBUTING.md .

At the moment there are large chunks and are commented out (as they do not apply this repo).

And there are several TODOs.

I think it would be nice to have this document up in the `dev` branch and then we can update it bit by bit.

## For reviewers

- Does this make sense ? Anything else that your past or present self is unclear that could help to have here.
- Anything that should be added in here? Maybe just as a TODO for now.

### Ideas

- Is there any content from the Turing-Way we can reuse?
- I think that once we have a working basis on this one, we could adapt it for the other repos (CPP_SPM, CPP_PTB): though something smarter would be to have a separate repo with only this document and all repos point to it. This way there is an organization wide documentation and we only have to keep one document updated and not one per repo.

### CODE REVIEW

**Review by CerenB — COMMENTED** [review]
Heya! I'm a bit confused, did you take the Contributing.md text from bids-matlab? and you want people follow the same PR format when they want to send a PR to cpp-bids? 

HOWTO make/runs tests part is definitely needed! 👍

**Comment by Remi-Gau** [comment]
> Heya! I'm a bit confused, did you take the Contributing.md text from bids-matlab? and you want people follow the same PR format when they want to send a PR to cpp-bids?

I took this from the bids-specification to adapt it for bids-matlab and while I was at it I figured I would use it for our own stuff as well.

The process for PR on BIDS matlab is going to be a bit more involved than our own, but I would like this document to have basic infos about PRs, how to open them, how to review them...

We don't have to have a super formal process for PRs beyond the `master` and `dev` thing. For now... :wink:

**Comment by CerenB** [comment]
it is up to you how things will evolve but my hope was not to be too serious so people would not be afraid of getting involved/contributing. and bids-PR structure is really good to read but a bit scary for me to contribute to their work. I'm still failing at miss_hit :D :D though the last time I bet it was due to CPP-SPM changes!

**Comment by Remi-Gau** [comment]
Yeah I agree that for our stuff we should just limit things to: send your PR to the dev branch. Which is most likely how it will be for bids-matlab in the end too I think.

**Comment by CerenB** [comment]
I think it's definitely helpful and structured well. Only we should add somewhere "don't be afraid, be nice when committing a PR".

**Comment by Remi-Gau** [comment]
And when reviewing it !!

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,381 @@
+# Contributing to the CPP-BIDS
+
+**Welcome to the BIDS-matlab repository!**
```
Comment (marcobarilari) [inline CONTRIBUTING.md:3]:
```suggestion
**Welcome to the CPP-BIDS repository!**
```

**Review by marcobarilari — APPROVED** [review]
Sorry for the super delay, I think it's great to have something like this. Personally, I feel more afraid of opening issues and prs in those repos where there is no invitation and/or 'guide' on how to do it. It is also very useful for us as a reminder cause, I admit, sometimes I am really lost in reviewing/have a look at some prs where there are little info on the changes.

I also agree that there should be one for the organization.

++++++1

**Comment by Remi-Gau** [comment]
Do you think we should add more info about forking and cloning and similar git magic.

Maybe a couple of tips from the bids-validator repo: https://github.com/bids-standard/bids-validator/blob/master/CONTRIBUTING.md#using-the-development-version-of-bids-validator

**Comment by marcobarilari** [comment]
Why not! Then let's move everything to an organization level from the beginning, WDYT?

**Comment by Remi-Gau** [comment]
> Why not! Then let's move everything to an organization level from the beginning, WDYT?

Sure I will close this, create a new repo and move everything there and tag you all there.

**Comment by Remi-Gau** [comment]
There is away to create organization wide documents on github: we'll use that options.

https://docs.github.com/en/free-pro-team@latest/github/building-a-strong-community/creating-a-default-community-health-file

**Comment by Remi-Gau** [comment]
Done. All in a repo called `.github`. Did not create a pull request. Figured that this way we have a basis we can start using, and improve incrementally.

https://github.com/cpp-lln-lab/.github/blob/main/CONTRIBUTING.md

The way it works is that now when you open issue or a PR there will be a link to the `contributing guidelines` that points this one document.

I suspect that we should also make an more visible mention of it in our READMEs and Documentation.


Screenshot 2020-11-09 21:30:58

**Comment by Remi-Gau** [comment]
Will therefore transform this PR into simply adding a link in the READM to point the other repo.

**Comment by marcobarilari** [comment]
if I understand correctly, these files will be added to any public repo in this org. Correct?

**Pull request merged**